### PR TITLE
Party feature

### PR DESCRIPTION
**Pull request by fsufyan**

### CODE REVIEW

**Comment by fsufyan** [comment]
@lugehorsam

File: Sources/Nakama/Mappings.swift [hunk 1]
```diff
@@ -333,3 +352,16 @@ extension Nakama_Realtime_UserPresence {
...
 }
+
+extension Nakama_Realtime_Party {
+    func toParty() -> Party {
+        return Party(
+            id: self.partyID,
+            open: self.open,
+            maxSize: Int(self.maxSize),
+            self_p: self.self_p.toUserPresence(),
+            Leader: self.leader.toUserPresence(),
```
Comment (lugehorsam) [inline Sources/Nakama/Mappings.swift:363]: is this meant to be uppercase?

File: Sources/Nakama/Socket.swift [hunk 2]
```diff
@@ -173,6 +201,10 @@ public final class Socket : SocketProtocol {
                             promise.fail(NakamaRealtimeError(error: error))
                         } else if let promise = collatedPromise as? EventLoopPromise<Nakama_Realtime_Status> {
                             promise.fail(NakamaRealtimeError(error: error))
+                        } else if let promise = collatedPromise as? EventLoopPromise<Nakama_Realtime_Party> {
+                            promise.fail(NakamaRealtimeError(error: error))
+                        } else if let promise = collatedPromise as? EventLoopPromise<SwiftProtobuf.Google_Protobuf_Empty> {
+                            promise.fail(NakamaRealtimeError(error: error))
```
Comment (lugehorsam) [inline Sources/Nakama/Socket.swift:207]: how come we have a different promise.Fail for every promise type? i.e., why not just once if there is an error?

**Review by lugehorsam — CHANGES_REQUESTED** [review]

File: Sources/Nakama/Mappings.swift (hunk 1 above)
Comment (fsufyan) [reply Sources/Nakama/Mappings.swift:363]: Yeah because self is a reserved keyword.

File: Sources/Nakama/Socket.swift (hunk 2 above)
Comment (fsufyan) [reply Sources/Nakama/Socket.swift:207]: You are right we could just simplify with checking for any error:
```
if let promise = collatedPromise as? EventLoopPromise<Any> {
    promise.fail(NakamaRealtimeError(error: error))
}
```

File: Sources/Nakama/Mappings.swift (hunk 1 above)
Comment (fsufyan) [reply Sources/Nakama/Mappings.swift:363]: Right, Leader should be lowercase.

**Review by lugehorsam — APPROVED** [review]

**Pull request merged**